cmd/opampsupervisor/supervisor/server.go

```diff
@@ -36,6 +44,11 @@ func newServerSettings(fs flattenedSettings) server.StartSettings {
 
 								return &protobufs.ServerToAgent{}
 							},
+							OnConnectedFunc: func(ctx context.Context, conn serverTypes.Connection) {
```

Comment:
Should this be?
```suggestion
							OnConnectionCloseFunc: func(ctx context.Context, conn serverTypes.Connection) {
```

Reply:
Yeah, good catch, totally should be the close function. Updated to correctly call the close func.